pkg/datapath/loader/netlink.go

```diff
@@ -70,7 +70,7 @@ func replaceDatapath(ctx context.Context, ifName, objPath, progSec, progDirectio
 
 	// Temporarily rename bpffs pins of maps whose definitions have changed in
 	// a new version of a datapath ELF.
-	if err := bpf.StartBPFFSMigration(bpf.GetMapRoot(), objPath); err != nil {
+	if err := bpf.StartBPFFSMigration(bpf.MapPrefixPath(), objPath); err != nil {
```

Comment:
I'm open to carving out the first fix (bpffs path) into a separate PR. I think it's rather critical given it can deadlock datapath regeneration (or even prevent agent start?) when maps get changed.

The second (deferred map removal) is mostly oriented towards 1.12, but I was considering the case where we need to add and backport a tail call as an 'emergency' if a complexity issue arises that we cannot solve any other way. It's completely up to you to; perhaps we should only backport this fix if/when such a situation actually arises.

Reply:
On second thought, I think this is good to go. I don't see this as a particularly invasive or risky change, it just defers cleanup to be slightly later.

Reply:
>  I think it's rather critical given it can deadlock datapath regeneration (or even prevent agent start?) when maps get changed.

Can you describe this in a bit more detail, also include it in the commit message? It's not immediately obvious to me which conditions would cause this kind of situation or even that the first patch is resolving such a serious bug.

> On second thought, I think this is good to go. I don't see this as a particularly invasive or risky change, it just defers cleanup to be slightly later.

:heavy_plus_sign: Assuming we resolve the other feedback, this should be fairly low risk.